phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -92,7 +100,7 @@ public MutationState(int maxSize, PhoenixConnection connection, long sizeOffset)
         this.sizeOffset = sizeOffset;
     }
     
```

Comment:
and this (part 2 of 3):

    // For tests to pass through a Map that maintains order (and keep this.mutations as final)
    public MutationState(int maxSize, PhoenixConnection connection, Map<ImmutableBytesPtr,RowMutationState> mutations) {
        this(maxSize,connection,0, mutations);
    }

    public MutationState(int maxSize, PhoenixConnection connection, long sizeOffset) {
        this.maxSize = maxSize;
        this.connection = connection;
        this.sizeOffset = sizeOffset;
        this.mutations = Maps.newHashMapWithExpectedSize(3); // TODO: size based on something else as this is a bad default
    }
    

    public MutationState(int maxSize, PhoenixConnection connection, long sizeOffset, Map<ImmutableBytesPtr,RowMutationState> mutations) {
        this.maxSize = maxSize;
        this.connection = connection;
        this.sizeOffset = sizeOffset;
        this.mutations = mutations;
    }